### PR TITLE
Add social flow

### PR DESCRIPTION
**Pull request by cmontella**

### CODE REVIEW

File: food-truck.eve.md [hunk 1]
```diff
@@ -352,19 +352,118 @@ bind @browser
...
-- Twitter
-- Instagram
-- All off by default
-- Clicking turns on that site for posting
+
+Draw the social media page. There are two cases we want to handle. If the truck isn't set up or there are no #enabled social media #integrations, then we should display a message prompting the owner to add those in.
+
+```
+search @browser @session
+  wrapper = [#page-wrapper page: "social-flow"]
```
Comment (joshuafcole) [inline food-truck.eve.md:360]: With an integration added, this banner is still (incorrectly) open when I visit the flow page until I click the toggle button for that integration in the posting flow.

File: food-truck.eve.md [hunk 2]
```diff
@@ -352,19 +352,118 @@ bind @browser
...
+    [#button #to-home text: "home"]
+    [#div children:
+      [#div text: "Add a message:"]
+      [#editable #social-message default: "message..."]]    
+    [#div #integrations children:
+      [#div text: "Add a photo:"]
+      [#image-container #social-image prompt: "photo..."]]
+    [#div children: 
+      [#div text: "Select social networks"]
+      [#span #integration integration class: "ion-social-{{integration.name}}{{selected?}}"]]
```
Comment (joshuafcole) [inline food-truck.eve.md:410]: If the social flow is left after clicking an integration, it remains visually toggled on but the issue with the banner above now happens with the icon's state inverted.

File: food-truck.eve.md [hunk 3]
```diff
@@ -436,12 +539,14 @@ search @browser
   [#truck-name value]
```
Comment (joshuafcole) [inline food-truck.eve.md:539]: This flow could use some UX love. De-focusing the input and hitting esc should do something here. Also, what does the edit button do here?

File: food-truck.eve.md [hunk 4]
```diff
@@ -352,19 +352,118 @@ bind @browser
...
+  (add, remove) = if integration = [#selected] then ("","selected")
+                  else ("selected", "")
+  
+  
+commit
+  integration.tag += add
+  integration.tag -= remove  
+```
+
+The "post" button is enabled when at least one social integration is selected, and a message or an image is uploaded
```
Comment (joshuafcole) [inline food-truck.eve.md:432]: There doesn't appear to be a visual indicator as to whether the post button is enabled or not. Assuming I managed to get it into an enabled state, there doesn't seem to be feedback for a successful posting. We should do a banner and clear the forms.

File: food-truck.eve.md (hunk 3 above)
Comment (joshuafcole) [reply food-truck.eve.md:539]: (This applies to all #editables, but they're not part of the changeset)

**Review by joshuafcole — DISMISSED** [review]

File: food-truck.eve.md (hunk 4 above)
Comment (cmontella) [reply food-truck.eve.md:432]: I add a class when it's enabled, which should provide the visual indication when the app is styled.

File: food-truck.eve.md (hunk 3 above)
Comment (cmontella) [reply food-truck.eve.md:539]: I'll fix this in a separate PR

**Pull request closed without merging**